provider/datagen/src/source.rs

```diff
@@ -390,16 +390,32 @@ impl AbstractFs {
                 } else {
                     return Ok(());
                 };
-                lazy_static::lazy_static! {
-                    static ref CACHE: cached_path::Cache = cached_path::CacheBuilder::new()
-                        .freshness_lifetime(u64::MAX)
-                        .progress_bar(None)
-                        .build()
-                        .unwrap();
-                }
-                let root = CACHE
-                    .cached_path(resource)
-                    .map_err(|e| DataError::custom("Download").with_display_context(&e))?;
+                #[cfg(feature = "networking")]
+                let root = {
+                    lazy_static::lazy_static! {
+                        static ref CACHE: cached_path::Cache = cached_path::CacheBuilder::new()
+                            .freshness_lifetime(u64::MAX)
+                            .progress_bar(None)
+                            .build()
+                            .unwrap();
+                    }
+
+                    CACHE
+                        .cached_path(resource)
```

Comment:
Not sure I understand this part of the code. Do both cldr-root and cldr-tag go through this same cached_path code path?

Reply:
Yes; because you can specify a URL to cldr-root

Reply:
I re-read the code and I'm not convinced that we ever hit the new code path. The line above is

```
                let resource = if let Err(resource) = lock.deref() {
                    resource
                } else {
                    return Ok(());
                };
```

so this is only hit when the lock contains an `Err`. The only way the lock gets that value seems to be in

```
    fn new_from_url(path: String) -> Self {
        Self::Zip(RwLock::new(Err(path)))
    }
```

whereas if you have a local path, you use `fn new`, which says

```
            Ok(Self::Zip(RwLock::new(Ok(ZipArchive::new(Cursor::new(
                std::fs::read(&root)?,
            ))
            .map_err(|e| {
                DataError::custom("Zip").with_display_context(&e)
            })?))))
```

Reply:
ah, hm